From b39fc478d6d1b6560a5d1bfd2110512562767e59 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 07 Oct 2020 22:01:31 +0300
Subject: [PATCH] [SW-21702] goya2: replace delay with register check

Currently, delay is used of 300 microseconds to wait for PHY init
and executing Synopsys phy WA. This patch replaces delay with
sram_init_done registers check to confirm PHY init followed by
executing Synopsys workaround.

Change-Id: I8b68f32bdc524ac2209953178810316347c8efdc
---

diff --git a/goya2/common.c b/goya2/common.c
index 43b7a05..99a7345 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -112,8 +112,10 @@
 	/* Disable the EQ REDO */
 	regmap_writel(0x00402801, mmPCIE_DBI_GEN3_RELATED_OFF);
 
-	/* delay before executing Synopsys W/A > 300us */
-	udelay(300);
+	/* Wait for init_done before executing W/A */
+	while (!(regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_0) &&
+		regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_1)))
+		;
 
 	/* Execute the Synopsys W/A */
 	regmap_writel(0xF, mmPCIE_PHY_CR_PARA_CTRL);
